packages/bvaughn-architecture-demo/src/contexts/SelectedFrameContext.tsx

```diff
@@ -0,0 +1,61 @@
+import { FrameId, PauseId } from "@replayio/protocol";
+import { createContext, PropsWithChildren, useContext, useEffect, useMemo, useState } from "react";
+import useLoadedRegions from "../hooks/useRegions";
+import { ReplayClientContext } from "shared/client/ReplayClientContext";
+import { TimelineContext } from "./TimelineContext";
+import { isPointInRegions } from "shared/utils/time";
+import { getPauseForExecutionPoint } from "../suspense/PauseCache";
+
+interface PauseAndFrameId {
+  pauseId: PauseId;
+  frameId: FrameId;
+}
+
+export interface SelectedFrameContextType {
+  selectedFrameId: PauseAndFrameId | null;
+  setSelectedFrameId(pauseAndFrameId: PauseAndFrameId | null): void;
+}
```

Comment:
I think we should split `selectedFrameId` into two values to avoid confusion:
```suggestion
export interface SelectedFrameContextType {
  frameId: FrameId | null;
  pauseId: PauseId | null;
  setSelectedFrame(
    pauseId: PauseId | null,
    frameId: FrameId | null,
  ): void;
}
```

Reply:
I combined these values so that the type makes it clear that we either have a `pauseId` and a `frameId` or we have neither (because having just one of these IDs doesn't make sense).

Reply:
I don't think the way it is structured _is_ clear though. Maybe renaming the main thing could help.